### PR TITLE
Conditionally choose notifyPropertyChange context based on Ember version

### PR DESCRIPTION
**Pull request by fsmanuel**
I cherry picked the changes from @achambers in #68 and updated `ember-compatibility-helpers` to `^1.2.6`.

### CODE REVIEW

File: addon/mixin.js [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
 import { readOnly } from '@ember/object/computed';
-import { defineProperty, getProperties, set, get } from '@ember/object';
+import { defineProperty, getProperties, set, get, notifyPropertyChange } from '@ember/object';
+
+import { gte, lte } from 'ember-compatibility-helpers';
 
 import { aliasMethod, empty } from './helpers';
 
-const { notifyPropertyChange, meta } = Ember; // eslint-disable-line ember/new-module-imports
+import { meta as metaFor } from '@ember/-internals/meta';
```
Comment (fsmanuel) [inline addon/mixin.js:12]: @lukemelia I don't know how much better it is to use the `-internals` import instead of the old `import Ember from 'ember';`

Comment (lukemelia) [reply addon/mixin.js:12]: Yeah, both seem not great. But then we're defining a mixin, which itself is not particularly modern so... 🤷

**Review by lukemelia — APPROVED** [review]

File: addon/mixin.js (hunk 1 above)
Comment (fsmanuel) [reply addon/mixin.js:18]: so true! 🤣

**Comment by lukemelia** [comment]
Released as 2.1.0

**Pull request merged**